src/uniprotkb/components/entry/tabs/genomic-coordinates/Overlapping.tsx

```diff
@@ -0,0 +1,147 @@
+
+import { GenomicEntry } from './types';
+
+type OverlappingProps = {
+  taxID: number;
+  chromosome?: string;
+  start: number;
+  end: number;
+  currentEntry: string;
+};
+
+const Overlapping = ({
+  taxID,
+  chromosome,
+  start,
+  end,
+  currentEntry,
+}: OverlappingProps) => {
+  const sp = new URLSearchParams({
+    taxid: `${taxID}`,
+    location: `${start}-${end}`,
+  });
+  if (chromosome) {
+    sp.set('chromosome', chromosome);
+  }
+
+  /** Fetch all entries for which part of the genome location overlaps (even
+   * partially) to the current one */
+  const { loading, data, error, status } = useDataApi<Array<GenomicEntry>>(
+    `${proteinsApi.coordinates()}?${sp}`
+  );
+
+  if (loading) {
+    return <SpinnerIcon width="1em" height="1em" />;
+  }
+
+  if (error && status !== 404) {
+    return <em>An error happened</em>;
+  }
+
+  // Removes the original protein from the list of proteins at these coordinates
+  const otherEntries = data?.filter(
+    (entry) => entry.accession !== currentEntry
+  );
+
+  if (!otherEntries?.length) {
+    return <>none</>;
+  }
+
+  // Gather minmax coordinates of overlapping entries
+  let min = start;
+  let max = end;
+  const entryList: JSX.Element[] = [];
+
+  // Render an entry link for each overlapping entry
+  for (const entry of otherEntries) {
+    const entryAndLocations = [
+      <Link
+        to={getEntryPath(
+          Namespace.uniprotkb,
+          entry.accession,
+          TabLocation.Entry
+        )}
+        key={entry.accession}
+      >
+        {entry.accession}
+      </Link>,
+      ' (',
+    ];
+    // Render an Ensembl link for each overlapping entry's overlapping location
+    for (const { genomicLocation } of entry.gnCoordinate || []) {
+      if (genomicLocation.chromosome !== chromosome) {
+        // This location not even within the original chromosome, skip it
+        continue; // eslint-disable-line no-continue
+      }
+      const otherStart = genomicLocation.reverseStrand
+        ? genomicLocation.end
+        : genomicLocation.start;
+      const otherEnd = genomicLocation.reverseStrand
+        ? genomicLocation.start
+        : genomicLocation.end;
+      if (otherStart === undefined || otherEnd === undefined) {
+        continue; // eslint-disable-line no-continue
+      }
+      const startWithinRange = otherStart >= start && otherStart <= end;
+      const endWithinRange = otherEnd >= start && otherEnd <= end;
+      const startBeforeRange = otherStart <= start;
+      const endAfterRange = otherEnd >= end;
+      if (
+        // No part within the range
+        !(startWithinRange || endWithinRange) &&
+        // This range does not include the current entry range
+        !(startBeforeRange && endAfterRange)
+      ) {
+        // This location not within the original location range, skip it
+        continue; // eslint-disable-line no-continue
+      }
+      if (entryAndLocations.length > 2) {
+        entryAndLocations.push(', ');
```

Comment:
When would we get here? I'm just thinking as we have so many commas already would a different separator help with visual parsing?

Reply:
Changed the logic in this area to have the logic finding entries and overlapping location in pure TS, and the JSX lower in the return statement.
Also, I'm now separating with ";" instead of ",".
I haven't found an example so far to check this though

Reply:
Found an example: Q9JIF9
Although that way you can have a look at what it looks like, this example is a bit strange because it has twice the same genomic location (with different transcript and translation IDs) and it's mapping to the exact same protein but unreviewed (see question in `#general` Slack channel)